src/Scenes/GameScene/GameScene.ts

```diff
@@ -30,30 +39,10 @@ export class GameScene implements Scene {
 		this._game.removeLayer(this._layer);
 	}
 
-	public update(): void {
-		if (this._game.keyboard.isKeyPressed('7')) {
-			this._session.runTurn(PlayerAction.MoveUpLeft);
-		} else if (this._game.keyboard.isKeyPressed('8')) {
-			this._session.runTurn(PlayerAction.MoveUp);
-		} else if (this._game.keyboard.isKeyPressed('9')) {
-			this._session.runTurn(PlayerAction.MoveUpRight);
-		} else if (this._game.keyboard.isKeyPressed('4')) {
-			this._session.runTurn(PlayerAction.MoveLeft);
-		} else if (this._game.keyboard.isKeyPressed('5')) {
-			this._session.runTurn(PlayerAction.Wait);
-		} else if (this._game.keyboard.isKeyPressed('6')) {
-			this._session.runTurn(PlayerAction.MoveRight);
-		} else if (this._game.keyboard.isKeyPressed('1')) {
-			this._session.runTurn(PlayerAction.MoveDownLeft);
-		} else if (this._game.keyboard.isKeyPressed('2')) {
-			this._session.runTurn(PlayerAction.MoveDown);
-		} else if (this._game.keyboard.isKeyPressed('3')) {
-			this._session.runTurn(PlayerAction.MoveDownRight);
-		} else if (this._game.keyboard.isKeyPressed('r')) {
-			const player = this._session.level.entities.getPlayer();
-			player && this._session.registerRecording(player.movesQueue.copy());
-			this._session.resetLevel();
-		}
+	public update(passedTime: number): void {
+		const controller = new SessionController(this, this._session);
```

Comment:
Is it right that we are creating a new SessionController every update?

Reply:
It's fine in that it's super cheap and I left it like that because there isn't much value in moving it to a field, and I also have the impression that at some point it'll be stateful and we'll want a clean state every update.

But we don't need it **right now**, so I've refactored it to be a field.